tests/testthat/test-beam.R

```diff
@@ -122,3 +165,341 @@ test_that("Distance equals range when beam is following the earth (elev = 0)", {
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp,
+      elev = 1:4,
+      distance = 90:100,
+      zlim = c(2, NA)
+    ),
+    regexp = "'zlim' should be a vector with two numeric values for upper and lower bound",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp,
+      elev = 1:4,
+      distance = 90:100,
+      zlim = c(42, 1)
+    ),
+    regexp = "'zlim' should be a vector with two numeric values for upper and lower bound",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp,
+      elev = 1:4,
+      distance = 80:90,
+      steps = c(4,2)
+    ),
+    regexp = "'step' should be a numeric value",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp,
+      elev = 1:4,
+      distance = 80:90,
+      steps = "a"
+    ),
+    regexp = "'step' should be a numeric value",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp,
+      elev = 1:4,
+      distance = 70:80,
+      quantity = "not_a_valid_quanity"
+    ),
+    regexp = "'quantity' should be one of 'dens' or 'eta'",
+    fixed = TRUE
+  )
+  example_vp_no_antenna <- example_vp
+  example_vp_no_antenna$attributes$where$height <- NULL
+  expect_error(
+    beam_profile_overlap(
+      example_vp_no_antenna,
+      elev = 1:4,
+      distance = 1:3
+    ),
+    regexp = "antenna height cannot be found in polar volume, specify antenna height using 'antenna' argument",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp_no_antenna,
+      elev = 1:4,
+      distance = 1:3,
+      antenna = c(4,2)
+    ),
+    regexp = "antenna is not a number (a length one numeric vector).",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp_no_antenna,
+      elev = 1:4,
+      distance = 1:3,
+      antenna = "not_a_number"
+    ),
+    regexp = "antenna is not a number (a length one numeric vector).",
+    fixed = TRUE
+  )
+  example_vp_no_lat <- example_vp
+  example_vp_no_lat$attributes$where$lat <- NULL
+  expect_error(
+    beam_profile_overlap(
+      example_vp_no_lat,
+      elev = 1:4,
+      distance = 1:3
+    ),
+    regexp = "radar latitude cannot be found in polar volume, specify using 'lat' argument",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_profile_overlap(
+      example_vp_no_lat,
+      elev = 1:4,
+      distance = 1:3,
+      lat = "b"
+    ),
+    regexp = "lat is not a number (a length one numeric vector).",
+    fixed = TRUE
+  )
+})
+
+test_that("beam_width() returns error on incorrect parameters", {
+  range <- seq(0, 100000, 100)
+  expect_error(
+    beam_width(c(range,"a")),
+    regexp = "range is not a numeric or integer vector",
+    fixed = TRUE
+  )
+  expect_error(
+    beam_width(range,
+               beam_angle = "a")
+  )
+})
+
+
+test_that("gaussian_beam_profile returns error on incorrect parameters", {
```

Comment:
Is this function actually used anywhere? It seems to be a wrapper for `gaussian_beam_profile_internal()` with better failstates. Is it possible this function was supposed to replace` gaussian_beam_profile_internal()` in other functions?